ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/MigrationsDirectoryUtil.java

```diff
@@ -99,4 +94,34 @@ public static String computeHashForFile(final String filename) {
           "Could not compute hash for file '%s': %s", filename, e.getMessage()));
     }
   }
+
+  public static String getNameFromMigrationFilePath(final String filename) {
+    return filename
+        .substring(filename.indexOf("__") + 2, filename.indexOf(".sql"))
+        .replace('_', ' ');
+  }
+
+  public static int getVersionFromMigrationFilePath(final String filename) {
+    return Integer.parseInt(filename.substring(1, 7));
+  }
+
+  public static List<Migration> getAllMigrations(final String migrationsDir) {
+    final File directory = new File(migrationsDir);
+    if (!directory.isDirectory()) {
+      throw new MigrationException(migrationsDir + " is not a directory.");
+    }
+
+    final String[] names = directory.list();
+    if (names == null) {
+      throw new MigrationException("Failed to retrieve files from " + migrationsDir);
+    }
+
+    return Arrays.stream(names)
+        .map(name ->new Migration(
+            getVersionFromMigrationFilePath(name),
+            getNameFromMigrationFilePath(name),
+            computeHashForFile(migrationsDir + "/" + name),
```

Comment:
Let's not load all the contents of every file and compute the hash for every file each time -- we should only do this for the files we actually need to do it for.